.github/workflows/ci.yml

```diff
@@ -11,6 +11,10 @@ on:
       - master
   workflow_dispatch:
 
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: ${{ github.head_ref != 'master' }}
```

Comment:
note: Same as in scarb
https://github.com/software-mansion/scarb/blob/741336e8abdcc2e13f852b65c5ce37ae2dd83a5a/.github/workflows/ci.yml#L10